sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisExternalCatalogSuite.scala

```diff
@@ -28,13 +29,28 @@ import org.apache.spark.sql.catalyst.expressions.{Alias, AttributeReference}
 import org.apache.spark.sql.catalyst.plans.logical.{LocalRelation, Project}
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
+import org.apache.spark.util.Utils
 
 class AnalysisExternalCatalogSuite extends AnalysisTest with Matchers {
+  var tempDir: File = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    tempDir = Utils.createTempDir()
+  }
+
+  override def afterEach: Unit = {
+    try {
+      Utils.deleteRecursively(tempDir)
+    } finally {
+      super.afterEach()
+    }
+  }
```

Comment:
How about creating a root temp dir here and each temp dir under the root instead of the current one?

Reply:
you mean to create and use as follows
```
val rootTempDir  = Utils.createTempDir()
val tempDir = Utils.createTempDir(rootTempDir)
```

Use this tempDir in each testcase and delete in `afterEach` ?

Reply:
We need to remove it for each test? I feel ok to remove `rootTempDir` recursively in `afterAll`.

Reply:
clean up should happen after each test case execution, so only I ve put in the forEach